std/experimental/interpolate.d

```diff
@@ -0,0 +1,222 @@
+//
+//
+// TODO: move into std/experimental/typecons.d
+//
+//
+
+/*
+!!!!!!!!!!!!!!!!!!!!!!!
+!!!!!!!!!!!!!!!!!!!!!!!
+!!!!!!!!!!!!!!!!!!!!!!!
+!!!!!!!!!!!!!!!!!!!!!!!
+Need to publicly import std.typecons : tuple because the mixin code
+generated by the interp function requires it.
+
+This does have the disadvantage that the following won't work:
+
+import std.typecons : interp;
+
+int a = 42;
+writeln(mixin(interp("a is $(a)")));
+
+// ERROR: unknown symbol 'tuple'
+*/
+public import std.typecons : tuple;
+
+/**
+Interpolates the given string into a code that creates a tuple of string literals and expressions.
+The code returned should be passed to `mixin` in order to create the tuple. The `tuple` symbol
+from `std.typecons` should also be imported since the code returned depends on this function.
+
+Example:
+---
+int a = 42;
+
+writeln(mixin(interp("a is $(a)")));
+// same as: writeln(mixin(`tuple("a is ", a).expand`));
+
+// Output:
+// a is 42
+---
+*/
+string interp(string str, string file = __FILE__, size_t line = __LINE__) pure @safe
+{
+    return "tuple(" ~ interpToCommaExpression(str, file, line) ~ ").expand";
```

Comment:
I think we could avoid the public import by (essentially) doing this:
```d
return q{() @property {import std.typecons : tuple; return tuple(...);}};
```

Reply:
great idea...changed